### PR TITLE
test: Lint Dockerfiles with Trivy

### PR DESCRIPTION
**Pull request by atombrella**
## Description

I think it'd be nice to use trivy on the bundled Dockerfiles. There were warnings in all of them, which is just about the missing `USER` directive. There was already one effort to do similar with Hadolint. I haven't investigated if Trivy or Hadolint are 1:1 with regards to rules, but it seems that both check for missing `USER` and dangling package lists (not really needed anyway), and not using the `latest` tag.

Since I fix warnings in the first commit, and then introduce the linter, perhaps using `rebase and merge` is best practice to use. I'm rather indifferent which one that 

There's no GitHub action for Trivy, only for Azure DevOps, as far as I could tell.

## Related PRs
- [x] #1121

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by atombrella** [comment]
@knqyf263 It appears to work. https://gist.github.com/atombrella/465b4c7d4ff628f1e1d97d8f9cec5d80 I made a typo with the `adduser` command, which is fixed now.

I think it'd be nice to at least use trivy (maybe with explicit ignoring the `USER` warning for the bundled Docker files) in the repository's pipeline.

**Comment by atombrella** [comment]
@knqyf263 I got the workflow running in my own fork, and then creating a PR. The paths for where GitHub mounts the working directory and related paths is a bit hidden in the documentation. Could you please trigger it here as well, thank you? There are no warnings (nor any output, which is perhaps a bit disconcerning), after the `USER` directive has been added.

**Comment by atombrella** [comment]
@knqyf263 I forgot the checkout step for the lint job :man_facepalming: It runs now. I've been working in my local fork, running the workflow from there.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 FROM alpine:3.17.0
-RUN apk --no-cache add ca-certificates git
+RUN apk --no-cache add ca-certificates git && \
```
Comment (knqyf263) [inline Dockerfile:2]: We used to tried switching non-root user, but it didn't work well.
https://github.com/aquasecurity/trivy/issues/580

**Pull request closed without merging**